tests/cephfs/cephfs_mirroring/cephfs_mirroring_utils.py

```diff
@@ -386,10 +386,25 @@ def validate_synchronization(
             sudo=True, cmd="ceph fs snapshot mirror daemon status -f json"
         )
         data = json.loads(out)
-        filesystem_id = data[0]["filesystems"][0]["filesystem_id"]
-        peer_uuid = data[0]["filesystems"][0]["peers"][0]["uuid"]
-        log.info(f"Filesystem ID: {filesystem_id}")
-        log.info(f"Peer UUID: {peer_uuid}")
+        log.info(f"Json output : {data}")
+        filesystem_info = {}
+        for filesystem in data[0]["filesystems"]:
+            filesystem_id = filesystem["filesystem_id"]
+            log.info(f"File ID is {filesystem_id}")
+            name = filesystem["name"]
+            log.info(f"name  is {name}")
+            peers = filesystem["peers"]
+            log.info(f"peers is {peers}")
+            for peer in peers:
+                peer_uuid = peer["uuid"]
+                log.info(f"peer uuid is {peer_uuid}")
+                if name == fs_name:
+                    filesystem_info[filesystem_id] = peer_uuid
```

Comment:
can you make get_peer_uuid as seperate function by accepting filesystem name as argument
get_peer_uuid(fs_name="cephfs")

Reply:
`log.info("Fetch the FSID of the ceph cluster")
def get_fsid(self, cephfs_mirror_node):
    out, _ = cephfs_mirror_node.exec_command(sudo=True, cmd="cephadm ls")
    data = json.loads(out)
    fsid = data[0]["fsid"]
    return fsid


log.info("Fetch the cephfs-mirror daemon name")
def get_daemon_name(self, source_clients):
    out, _ = source_clients.exec_command(sudo=True, cmd="ceph orch ps --daemon_type cephfs-mirror -f json")
    data = json.loads(out)
    daemon_name = data[0]["daemon_name"]
    return daemon_name


log.info("Fetch the Filesystem Info to identify the Peer ID of the Peer Connection")
def get_filesystem_info_for_fetching_peer_id(self, source_clients, fs_name):
    out, _ = source_clients.exec_command(sudo=True, cmd="ceph fs snapshot mirror daemon status -f json")
    data = json.loads(out)
    filesystem_info = {}
    for filesystem in data[0]["filesystems"]:
        filesystem_id = filesystem["filesystem_id"]
        name = filesystem["name"]
        peers = filesystem["peers"]
        for peer in peers:
            peer_uuid = peer["uuid"]
            if name == fs_name:
                filesystem_info[filesystem_id] = peer_uuid
    return filesystem_info

def get_asok_file(self, cephfs_mirror_node, fsid, daemon_name):
    cmd = f"cd /var/run/ceph/{fsid}/ ; ls -1tr ceph-client.{daemon_name}* | head -n 1 | tr -d '\\n'"
    file = cephfs_mirror_node.exec_command(sudo=True, cmd=cmd)
    asok_file = file[0].replace("\\n", "")
    return asok_file

`

Reply:
@AmarnatReddy = Shall I split like this ?

Reply:
def validate_synchronization(self, cephfs_mirror_node, source_clients, fs_name, snap_count):
    fsid = self.get_fsid(cephfs_mirror_node)
    daemon_name = self.get_daemon_name(source_clients)
    filesystem_info = self.get_filesystem_info(source_clients, fs_name)
    for filesystem_id, peer_uuid in filesystem_info.items():
    asok_file = self.get_asok_file(cephfs_mirror_node, fsid, daemon_name)
    
    return fsid, asok_file, filesystem_id, peer_uuid